src/server/journal/journal_slice.cc

```diff
@@ -142,34 +142,36 @@ std::string_view JournalSlice::GetEntry(LSN lsn) const {
   return (*ring_buffer_)[lsn - start].data;
 }
 
-void JournalSlice::AddLogRecord(const Entry& entry, bool await) {
+void JournalSlice::SetFlushToSink(bool allow_flush) {
+  enable_journal_flush_ = allow_flush;
```

Comment:
I suggest to a least check-fail on recursive calls, i.e. 
`CHECK(allow_flush || enable_journal_flush_)`